src/domain/common/usecase/SelectDataUseCase.js

```diff
@@ -1,11 +1,25 @@
+const { NoMatchingElementsFoundError, InvalidSelectorDataError } = require('../utils/error');
+
 class SelectDataUseCase {
   constructor(dataSelectorFinder, logger) {
     this.dataSelectorFinder = dataSelectorFinder;
     this.logger = logger;
   }
 
   async selectFromRawData(fetchedData, contentType, path) {
-    const selectedData = this.dataSelectorFinder.find(contentType).select(fetchedData, path);
+    let selectedData;
+
+    try {
+      selectedData = await this.dataSelectorFinder.find(contentType).select(fetchedData, path);
+    } catch (e) {
```

Comment:
This could be taken under discussion. Because until the rest of the implementation is proper, then only invalid data can be thrown here. But I also see your point about hiding potential problems other than that. @tcichowicz @przemyslawjanpietrzak what do you think?